pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/TNCudaBase.hpp

```diff
@@ -141,6 +155,127 @@ class TNCudaBase : public TensornetBase<Precision, Derived> {
         return dev_tag_;
     }
 
+    /**
+     * @brief Append multiple gates to the compute graph.
+     * NOTE: This function does not update the quantum state but only appends
+     * gate tensor operator to the graph.
+     * @param ops Vector of gate names to be applied in order.
+     * @param ops_wires Vector of wires on which to apply index-matched gate
+     * name.
+     * @param ops_adjoint Indicates whether gate at matched index is to be
+     * inverted.
+     */
+    void
+    applyOperations(const std::vector<std::string> &ops,
+                    const std::vector<std::vector<std::size_t>> &ops_wires,
+                    const std::vector<bool> &ops_adjoint,
+                    const std::vector<std::vector<PrecisionT>> &ops_params) {
+        const std::size_t numOperations = ops.size();
+        PL_ABORT_IF_NOT(
+            numOperations == ops_wires.size(),
+            "Invalid arguments: number of operations, wires, and inverses "
+            "must all be equal");
+        PL_ABORT_IF_NOT(
+            numOperations == ops_adjoint.size(),
+            "Invalid arguments: number of operations, wires and inverses"
+            "must all be equal");
+        for (std::size_t i = 0; i < numOperations; i++) {
+            this->applyOperation(ops[i], ops_wires[i], ops_adjoint[i],
+                                 ops_params[i]);
```

Comment:
Can we relax `ops_params` to be able to pass an empty vector for parameters as the default?

Reply:
yes, we can and this is what the line 212 does.